### DISCUSSION TITLE
Get the pointer for the graphicsAPI's texture

### DISCUSSION THREAD

**Discussion by ErikToth97** [google/filament#4603] [Q&A]
I need to get the pointer to the `graphicsAPI `texture for OpenVR. We have a plugin for OpenVR which only needs two texture pointers(for the two eyes) and handles everything else for VR. Unfortunately, these must be the `graphicsAPI`'s pointers so in case of `OpenGL `the `glTextureID`, but filament does not provide a way to access that.

What is the best way to get it (with modifying filament if it is needed)?

Another approach was to create an `OpenGL Texture `and import it into a `filament texture` after providing a `sharedGlContext  `for `filament engine,` but I was unable to do that because when I call glGenTextures I got and error that it is already defined in `OpenGL `(and bluegl) because both of them are linked, but I cannot unlink either of them because then I get undefined reference errors. And after further thought, we concluded that this path would come with other different problems.

**Comment by bejado** [comment]
Getting the OpenGL texture ID won't be much help because the ID will be unique to Filament's OpenGL context, which is associated with our background render thread.

Are you on Windows? #4523 should have fixed the linking errors you're seeing.
